packages/go_router/lib/src/delegate.dart

```diff
@@ -94,14 +95,18 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
   }
 
   /// Pushes the given location onto the page stack
-  void push(RouteMatchList matches) {
+  void push(RouteMatchList matches, {PageKey? keyToUse}) {
     assert(matches.last.route is! ShellRoute);
 
-    // Remap the pageKey to allow any number of the same page on the stack
-    final int count = (_pushCounts[matches.fullpath] ?? 0) + 1;
-    _pushCounts[matches.fullpath] = count;
-    final ValueKey<String> pageKey =
-        ValueKey<String>('${matches.fullpath}-p$count');
+    final PageKey pageKey;
+    if (keyToUse?.path == matches.fullpath) {
```

Comment:
why do we need this check? why not just use keyToUse if it is not null?

Reply:
Because if the page is different and the key doesn't match, ex: `/families/:fid` and `/families`, it should reuse it, right?

I could have named `keyToUse` `keyToReuseIfPossible` to be more explicit but it was a bit long